src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Services/AssemblyLoadContextWrapper.cs

```diff
@@ -31,7 +24,7 @@ private AssemblyLoadContextWrapper(AssemblyLoadContext assemblyLoadContext, Immu
             _logger = logger;
         }
 
-        public static bool TryLoadExtension(string assemblyFilePath, ILogger? logger, [NotNullWhen(true)] out Assembly? assembly)
+        public static bool TryLoadExtension(string assemblyFilePath, ILogger logger, [NotNullWhen(true)] out Assembly? assembly)
```

Comment:
Why does this even take a ILogger if we also passed a logger into this class in the constructor?

Reply:
This is static - there is a wrapper instance for each extension, which gets its logger from the parameter here.